### PR TITLE
rosbag2_storage_mcap: merge into rosbag2 repo

### PR DESCRIPTION
**Pull request by james-rms**
Merges `rosbag2_storage_mcap` into the `rosbag2` monorepo.

This PR merges the entire `main` branch of rosbag2_storage_mcap into `rolling`. Once this is approved, subsequent PRs into `galactic`, `foxy`, and `humble` will follow with the exact same code. The `rosbag2_storage_mcap` codebase is designed to be compatible with all active distributions, and we intend to keep the copies of the code in each release branch exactly the same for as long as possible.

Fixes https://github.com/ros-tooling/rosbag2_storage_mcap/issues/63

### CODE REVIEW

**Comment by amacneil** [comment]
Not sure if you are planning to squash or use merge commit, but if you are going to merge and keep the history around, it might be worth doing a bulk rename of issue references in commits to link fully qualified `https://github.com/ros-tooling/rosbag2_storage_mcap/issues/123` instead of just `#123` (since the latter will now point to the rosbag2 repo).

File: .github/workflows/lint.yml
Comment (MichaelOrlov) [inline .github/workflows/lint.yml:66]: Curious about why `shared_queues_vendor` was removed from the list?

File: mcap_vendor/package.xml
Comment (MichaelOrlov) [inline mcap_vendor/package.xml:7]: Please add at least
```
<maintainer email="ros-tooling@googlegroups.com">ROS Tooling Working Group</maintainer>
```
for maintainers

File: rosbag2_storage_mcap/package.xml
Comment (MichaelOrlov) [inline rosbag2_storage_mcap/package.xml:7]: Please add at least
```
<maintainer email="ros-tooling@googlegroups.com">ROS Tooling Working Group</maintainer>
```
for maintainers

**Review by MichaelOrlov — COMMENTED** [review]
@james-rms It looks good to me among a few nitpicks.
I assume that this migration is supposed to be pretty much straight forward copy-past right?
If it was some changes in baseline files could you please highlight them in MR description?

File: .github/workflows/lint.yml
Comment (james-rms) [reply .github/workflows/lint.yml:66]: Good catch, that was an oversight. I originally removed `shared_queues_vendor` when I split up the packages into a matrix, because then each package got its own lint job and `shared_queues_vendor` failed because there's no code in that package to lint. I can re-add it now for completeness, even though it does nothing in this context.

**Comment by ros-discourse** [comment]
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/ros-2-tsc-meeting-minutes-2022-11-17/28457/1

File: rosbag2_storage_mcap/src/message_definition_cache.cpp
Comment (MichaelOrlov) [inline rosbag2_storage_mcap/src/message_definition_cache.cpp:86]: This is a good example why I prefer clang-format instead of the ~uglify~, sorry uncrustify

**Comment by emersonknapp** [comment]
I'd lean toward squash with a pointer to the original repo for those who want to look at the history. But, not a strong preference.

Is this actually ready for merge, or are we still doing a few things at the source? If so, should we mark this as draft for now?

File: .github/workflows/lint.yml
Comment (emersonknapp) [inline .github/workflows/lint.yml:68]: Feel free to add an extra `clang-format` block here.

**Comment by james-rms** [comment]
@emersonknapp should be ready to merge now - I wanted to wait to integrate some of the most recent PRs into https://github.com/ros-tooling/rosbag2_storage-mcap , but that's done now.

**Review by jhurliman — APPROVED** [review]

**Review by emersonknapp — APPROVED** [review]

**Review by amacneil — APPROVED** [review]
🚀

**Comment by MichaelOrlov** [comment]
@james-rms We should have been running full CI before merge. 
I hope we didn't brake anything.

**Comment by ros-discourse** [comment]
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/ros-2-tsc-meeting-minutes-2022-12-15/28840/1

**Pull request merged**
